sympy/algebras/quaternion.py

```diff
@@ -697,3 +697,17 @@ def to_rotation_matrix(self, v=None):
 
             return Matrix([[m00, m01, m02, m03], [m10, m11, m12, m13],
                           [m20, m21, m22, m23], [m30, m31, m32, m33]])
+
+    def differentiate(self, vector):
```

Comment:
Vector here is the angular velocity. I think it would be easier to follow if you called it that both here and in the docstring.

Reply:
This shouldn't be called differentiate. It is the rhs of the quaternion differential equation. Not sure what a better name than that would be...